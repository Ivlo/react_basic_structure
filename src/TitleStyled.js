import styled from "styled-components";
import media from "./styles/media";

// eslint-disable-next-line import/prefer-default-export
export const TitleStyled = styled.h1`
  margin-top: ${(props) => props.theme.spacing.medium};
  font-size: ${(props) => props.theme.font.size.large};
  border: 1px solid ${(props) => props.theme.colors.primary};
  @media only screen and ${media.device.medium} {
    border: 1px solid ${(props) => props.theme.colors.white};
  }
`;
