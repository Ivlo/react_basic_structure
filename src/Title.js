import React from "react";
import { TitleStyled } from "./TitleStyled";

const Title = ({ children, size }) => <TitleStyled size={size}>{children}</TitleStyled>;
export default Title;
