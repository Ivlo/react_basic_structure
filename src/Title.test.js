import React from "react";
import { shallow } from "enzyme";
import Title from "./Title";
import { TitleStyled } from "./TitleStyled";

const title = "Test Title";
const wrapper = shallow(<Title size="ivlo">{title}</Title>);

describe("Title", () => {
  it("should render the Title Component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders the Titles children", () => {
    expect(wrapper.find(TitleStyled).prop("size")).toEqual("ivlo");
    // expect(wrapper.find("h1").text()).toEqual(title);
  });
  it("renders the Titles children", () => {
    expect(wrapper.find(TitleStyled).text()).toEqual(title);
  });
});
