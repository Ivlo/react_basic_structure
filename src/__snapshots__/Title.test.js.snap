// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render the Title Component correctly 1`] = `
<styled.h1
  size="ivlo"
>
  Test Title
</styled.h1>
`;
